File: src/main/java/com/github/njuro/jard/utils/validation/PropertyValidationException.java
package com.github.njuro.jard.utils.validation;

import java.io.Serial;
import javax.validation.ValidationException;
import lombok.Getter;
import org.springframework.validation.BindingResult;

/**
 * Exception to be thrown when business validation of request object (such as form) fails. Handled
 * by {@link ValidationExceptionHandler}.
 */
public class PropertyValidationException extends ValidationException {

  @Serial private static final long serialVersionUID = 8579539307467463861L;

  @Getter private final transient BindingResult bindingResult;

  public PropertyValidationException(BindingResult bindingResult) {
    this.bindingResult = bindingResult;
  }

  public PropertyValidationException(String message) {
    super(message);
    bindingResult = null;
  }
}

File: src/main/kotlin/com/github/njuro/jard/rewrite/utils/validation/PropertyValidationException.kt
package com.github.njuro.jard.rewrite.utils.validation

import org.springframework.validation.BindingResult
import javax.validation.ValidationException

/**
 * Exception to be thrown when business validation of request object (such as form) fails. Handled
 * by [ValidationExceptionHandler].
 */
class PropertyValidationException : ValidationException {

    val bindingResult: BindingResult?

    constructor(bindingResult: BindingResult): super() {
        this.bindingResult = bindingResult
    }

    constructor(message: String): super(message) {
        this.bindingResult = null
    }
}

